zingolib/src/wallet/transaction_records_by_id.rs

```diff
@@ -206,9 +210,99 @@ impl TransactionRecordsById {
             });
         });
     }
-}
 
-impl TransactionRecordsById {
+    /// Calculate the fee for a transaction in the wallet
+    ///
+    /// # Error
+    ///
+    /// Returns [`crate::wallet::error::FeeError::ReceivedTransaction`] if no spends or outgoing_tx_data were found
+    /// in the wallet for this transaction, indicating this transaction was not created by this spend capability.
+    /// Returns [`crate::wallet::error::FeeError::SpendNotFound`] if any shielded spends in the transaction are not
+    /// found in the wallet, indicating that all shielded spends have not yet been synced. Also returns this error
+    /// if the transaction record contains outgoing_tx_data but no spends are found.
+    /// If a transparent spend has not yet been synced, the fee will be incorrect and return
+    /// [`crate::wallet::error::FeeError::FeeUnderflow`] if an underflow occurs.
+    /// The tracking of transparent spends will be improved on the next internal wallet version.
+    pub fn calculate_transaction_fee(
+        &self,
+        transaction_record: &TransactionRecord,
+    ) -> Result<u64, FeeError> {
+        let sapling_spends = transaction_record
+            .spent_sapling_nullifiers()
+            .iter()
+            .map(|nullifier| {
+                self.values()
+                    .flat_map(|transaction_record| transaction_record.sapling_notes())
+                    .find(|&note| {
+                        if let Some(nf) = note.nullifier() {
+                            nf == *nullifier
+                        } else {
+                            false
+                        }
+                    })
+                    .ok_or(FeeError::SpendNotFound)
```

Comment:
this error type should maybe include the nullifier?

Reply:
This comment would be addressed if this code:

https://github.com/Oscar-Pepper/zingolib/pull/13/files

were merged.